contracts/container/contract.go

```diff
@@ -2274,3 +2289,198 @@ func attributeFromBytes(b []byte) (Attribute, string) {
 
 	return res, ""
 }
+
+// SetAttribute sets container attribute. Not all container attributes can be changed
+// with SetAttribute. The supported list of attributes:
+// - CORS
+//
+// SetAttribute must have chain's committee multi-signature witness. Invocation
+// script, verification script and session token parameters are owner
+// credentials. They are transmitted in notary transactions carrying original
+// users' requests. IR verifies requests and approves them via multi-signature.
+// GetInfo throws:
+// - [cst.NotFoundError]
+// - [cst.ErrorEmptyName]
+// - [cst.ErrorEmptyValue]
+// - [cst.ErrorAlphabetAndOwnerCheckWitnessFailed]
+// - [cst.ErrorAttributeIsImmutable]
+// - Other errors, which depend on the corresponding attribute.
+func SetAttribute(cID interop.Hash256, name, value string) {
+	if name == "" {
+		panic(cst.ErrorEmptyName)
+	}
+
+	if value == "" {
+		panic(cst.ErrorEmptyValue)
+	}
+
+	var (
+		ctx  = storage.GetContext()
+		info = getInfo(ctx, cID)
+	)
+
+	if !runtime.CheckWitness(info.Owner) && !runtime.CheckWitness(common.AlphabetAddress()) {
+		panic(cst.ErrorAlphabetAndOwnerCheckWitnessFailed)
+	}
+
+	switch name {
+	case corsAttributeName:
+		setCORSAttribute(ctx, cID, info, value)
+	default:
+		panic(cst.ErrorAttributeIsImmutable)
+	}
+}
+
+// setCORSAttribute sets container CORS attribute.
+// Pass an empty list to empty the attribute.
+func setCORSAttribute(ctx storage.Context, cID interop.Hash256, info Info, payload string) {
+	var (
+		obj  = std.JSONDeserialize([]byte(payload))
+		list = obj.([]map[string]any)
+	)
+
+	for i, item := range list {
+		if err := validateCORSRule(item); err != "" {
+			panic("Rule:" + std.Itoa10(i) + ", " + err)
+		}
+	}
+
+	var (
+		exists bool
+	)
+
+	for i, attr := range info.Attributes {
+		if attr.Key == corsAttributeName {
+			exists = true
+			info.Attributes[i].Value = payload
+			break
+		}
+	}
+
+	if !exists {
+		info.Attributes = append(info.Attributes, Attribute{
+			Key:   corsAttributeName,
+			Value: payload,
+		})
+	}
+
+	cnrBytes := toBytes(info)
+
+	storage.Put(ctx, append([]byte{infoPrefix}, cID...), std.Serialize(info))
+	storage.Put(ctx, append(append([]byte{ownerKeyPrefix}, info.Owner...), cID...), cID)
+	storage.Put(ctx, append([]byte{containerKeyPrefix}, cID...), cnrBytes)
+}
+
+func validateCORSRule(rule map[string]any) string {
+	if len(rule) == 0 {
+		return "CORS rules is empty"
+	}
+
+	allowedMethods, ok := rule["AllowedMethods"]
+	if !ok {
+		return "AllowedMethods must be defined"
+	}
+	if err := validateCORSAllowedMethods(allowedMethods.([]any)); err != "" {
+		return err
```

Comment:
```suggestion
		return "invalid AllowedMethods: " + err
```
and everywhere below

Reply:
too many excess details: 
```
invalid rule #0: invalid AllowedMethods: invalid method METHOD
```

Old is understandable as well. Especially, we have reference to the rule position
```
invalid rule #0: invalid method METHOD
```

I leave as is

Reply:
with improved messages inside funcs, this is ok